dbms/src/TiDB/Schema/SchemaBuilder.cpp

```diff
@@ -895,50 +895,61 @@ void SchemaBuilder<Getter, NameMapper>::syncAllSchema()
     /// Create all databases.
     std::vector<DBInfoPtr> all_schemas = getter.listDBs();
 
-    // TODO:make parallel to speed up
     std::unordered_set<String> db_set;
 
-    for (const auto & db : all_schemas)
-    {
-        std::shared_lock<std::shared_mutex> shared_lock(shared_mutex_for_databases);
-        if (databases.find(db->id) == databases.end())
-        {
-            shared_lock.unlock();
-            applyCreateSchema(db);
-            db_set.emplace(name_mapper.mapDatabaseName(*db));
-            LOG_DEBUG(log, "Database {} created during sync all schemas", name_mapper.debugDatabaseName(*db));
-        }
-    }
+    //We can't use too large default_num_threads, otherwise, the lock grabbing time will be too much.
+    size_t default_num_threads = std::max(4UL, std::thread::hardware_concurrency());
+    auto sync_all_schema_thread_pool = ThreadPool(default_num_threads, default_num_threads / 2, default_num_threads * 2);
```

Comment:
I think `queue_size_` should be larger or equal to size of `all_schemas`, otherwise `sync_all_schema_wait_group->schedule(task);` will be blocked. Check `ThreadPoolImpl<Thread>::scheduleImpl`.

Reply:
Seems it will just block when the current scheduled jobs' num is larger than `queue_size_`. So all tasks will finally be scheduled.

Reply:
I think queue_size_ < all_schemas.size() is ok.